### DISCUSSION TITLE
Using Next.js `res.revalidate` when calling from a Vercel cron job does not work

### DISCUSSION THREAD

**Discussion by ikapo** [vercel/community#1655] [Help]
### Summary

I have an api route that calls `res.revalidate`, exactly like on https://nextjs.org/docs/basic-features/data-fetching/incremental-
static-regeneration#on-demand-revalidation.

When calling this API route (in my production deployment) via `curl` in CLI or `fetch`, the page is regenerated and returns new data, as it normally should.

When calling this route from a Vercel cron job though, the page is still regenerated (confirmed via logs) but the website still returns the stale data.
Maybe this is some caching issue with the request cron jobs send?
I'm on Next 13.2.0

**Comment by amyegan** [comment]
Hi Ikapo! Are you able to share your repo or a minimal reproduction so we can get a better look at what is going wrong?

**Comment by ikapo** [comment]
Sure! https://github.com/ikapo/minimal-repro-vercel-cron

After some investigation this seems to be related to DNS.
https://minimal-repro-vercel-cron-7r3ufdra1-ikapo.vercel.app/ Revalidates fine from calling the cron job,
but https://minimal-repro-vercel-cron.vercel.app/ stays stale

I have also tried creating an edge-runtime function which calls the revalidate route which works, but an api call which calls another api route seems unnecessary

**Comment by ikapo** [comment]
So I've managed to find a solution for this, but I think the docs are missing some information on ISR.

When deploying to Vercel with `getStaticProps`, the build artifacts created are unique to each domain, so when calling `res.revalidate` on `myapp-[randomid]-[username].vercel.app`, it will revalidate that domain only, and not any other domain you have associated with that deployment (e.g. `myapp.vercel.app`).

Cron jobs (currently beta) run under the `[appname]-[randomid]-[username].vercel.app` domain, and (currently) there is no way to customize that, so if you want to use them to call `/api/revalidate`, it will only revalidate that domain and not any other one you have associated with the deployment.

There is a solution to this, using rewrites.
In your `vercel.json`:
```json
{
  "crons": [
    {
      "path": "/api/revalidateMyDomain",
      "schedule": "[YOUR CRON SCHEDULE]"
    }
  ],
  "rewrites": [
    {
      "source": "/api/revalidateMyDomain",
      "destination": "https://[DOMAIN YOU WANT REVALIDATED]/api/revalidate"
    }
  ]
}

```
This way the cron job will be re-written to call the revalidation route on the domain you want.

I've confirmed this to also work with URL parametes like `/api/revalidateMyDomain?secret=abc` and they will reach the `/api/revalidate` route correctly after the rewrite.

@amyegan This revalidate-per-domain is not mentioned in the ISR docs and I think adding it will remove any confusion.
